sympy/simplify/trigsimp.py

```diff
@@ -426,12 +426,55 @@ def analyse_gens(gens, hints):
 _trigs = (TrigonometricFunction, HyperbolicFunction)
 
 
-def trigsimp(expr, **opts):
+def _trigsimp_inverse(rv):
+
+    funcs = [(sin, asin), (cos, acos), (sec, asec), (csc, acsc),
+             (tan, atan), (cot, acot)]
+
+    def check_args(x, y):
+        try:
+            return x.args[0] == y.args[0]
+        except IndexError:
+            return False
+
+    def f(rv):
+        # for simple functions
+        for func, funcinv in funcs:
+            if isinstance(rv, funcinv) and isinstance(rv.args[0], func):
+                return f(rv.args[0].args[0])
+
+        # for atan2 simplifications, harder because atan2 has 2 args
+        if type(rv) is atan2:
+            y, x = rv.args
+            if _coeff_isneg(y):
```

Comment:
Actually, we need to know the sign of x and y, not whether they have a negative sign. If we don't know the sign then we can't return the correct form, right? So part of `check_args` would be to see that `all(sign(i) in (-1,0,1) for i in (x, y))` and if not, return False. Then instead of using `_coeff_isneg` just use `sign` and take appropriate action for the following types of expressions: `[Dummy(positive=True), Dummy(zero=True), Dummy(negative=True)]`.

Reply:
Not really in this case! I'm supposing `x` and `y` are any kind of symbolic values. When I set, for example, $\operatorname{atan_2}(-y, x) = -\operatorname{atan_2}(y, x)$, this is valid both for positive and negative `y`.

Reply:
The same is valid for  $\operatorname{atan_2}(y, -x) = \pi - \operatorname{atan_2}(y, x)$, minus the set problems.